src/main/java/com/example/demo/dao/GradeLocalRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.dao;
+
+import com.example.demo.dao.entity.Grade;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class GradeLocalRepository implements GradeRepository {
+    private Map<Long, List<Grade>> gradeRegister;
+
+    public GradeLocalRepository() {
+        this.gradeRegister = new HashMap<>();
+    }
+
+    @Override
+    public List<Grade> getAllGrades() {
+        List<Grade> allGradesList = new ArrayList<>();
+        gradeRegister.values().stream()
+                .forEach(list -> allGradesList.addAll(list));
+        return allGradesList;
+    }
+
+    @Override
+    public List<Grade> getStudentGrades(long studentId) {
+        List<Grade> studentGradeList;
+        studentGradeList = gradeRegister.getOrDefault(studentId, List.of());
+        return studentGradeList;
+    }
+
+    @Override
+    public void addGrade(Grade grade) {
+        if (gradeRegister.containsKey(grade.getStudentId())) {
+            this.gradeRegister.get(grade.getStudentId()).add(grade);
+        } else {
+            List<Grade> newStudentGradeList = new ArrayList<>();
+            newStudentGradeList.add(grade);
+            this.gradeRegister.put(grade.getStudentId(), newStudentGradeList);
+        }
+    }
+
+    @Override
+    public boolean updateGrade(Grade updatedGrade, int oldGradeValue, int oldGradeWeight) {
```

Comment:
Podobna sytuacja jak w repo studenta mylacy typ zwracany i nazwa. A co jesli dwie oceny w liscie beda mialy tą samą oldGradeValue? Ty podmieniasz pierwszą, a nie konkretną (to pokazuje, że student bez problemu moze przechowywac liste ocen, bo nie ma znaczenia którą piątkę zaktualizowałes - wziąłeś pierwszą)

Reply:
robiąc update oceny, powinnismy podać ID tej oceny, oraz nową wartość. Po co całe `Gradle` oraz `oldGradeValue`? 

Chyba że uznajesz że oceny nie identyfikowalne, czyli jak uczeń ma cztery piątki, to ich nie rozróżniasz. Wydaje mi się że jednak chcesz je rozróżnić, bo potem będizesz chciał dopisać datę dodania oceny, z czego pochodzi (sprawdzian, kartkówka itd)

Reply:
Właśnie wydaje mi się że oceny nie mają swojego ID, tylko ID studenta do którego są przypisane. Jeśli ocena rzeczywiście ma znaczenie, ID dla oceny to strzał w 10tke :)

Reply:
Ocenki mają teraz swoje własne, osobne Id. To chyba rzeczywiście najlepsze podejście :P